File: tests/csharp/Reference.Tests.CSharp.xUnit/SimpleUnitTest.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;
using Xunit;
// ReSharper disable InconsistentNaming
// ReSharper disable xUnit1004

namespace Reference.Tests.CSharp.xUnit
{
    /// <summary>
    /// Simple Unit Test
    /// </summary>
    public class SimpleUnitTest
    {
        /// <summary>
        /// Ok Test
        /// </summary>
        [Fact]
        public void OkTest()
        {
            var str01 = "Hello World";
            var str02 = "hello world";

            Assert.Equal(str01, str02, ignoreCase: true);
        }

        /// <summary>
        /// Error test
        /// </summary>
        [Fact]
        public void ErrorTest()
        {
            var base64 = Convert.ToBase64String(new byte[] { 0x01, 0x02, 0x03, 0xA0, 0xA1, 0xA2, 0xA3 });
            StackLevel01(base64);
        }
        void StackLevel01(string base64)
        {
            StackLevel02(base64);
        }
        void StackLevel02(string base64)
        {
            StackLevel03(base64);
        }
        void StackLevel03(string base64)
        {
            Convert.FromBase64String(base64.Substring(1));
        }

        /// <summary>
        /// Skipped Test
        /// </summary>
        [Fact(Skip = "Skipped test demo")]
        public void SkipTest()
        {
        }

        /// <summary>
        /// Fail Test
        /// </summary>
        [Fact]
        public void FailTest()
        {
            Assert.True(IsPrime(411));

            bool IsPrime(int value)
            {
                if ((value & 1) == 0)
                    return value == 2;

                for (var i = 3; i * i <= value; i += 2)
                {
                    if (value % i == 0)
                        return false;
                }
                return value != 1;
            }
        }
    }
}

File: tests/vb/Reference.Tests.VB.xUnit/SimpleUnitTest.vb
Imports System.Net.Http
Imports System.Threading.Tasks
Imports Xunit
' ReSharper disable InconsistentNaming

''' <summary>
''' Simple Unit Test
''' </summary>
Public Class SimpleUnitTest

    ''' <summary>
    ''' Ok Test
    ''' </summary>
    <Fact>
    Public Sub OkTest()
        Dim str01 = "Hello World"
        Dim str02 = "hello world"

        Assert.Equal(str01, str02, ignoreCase:=True)
    End Sub

    ''' <summary>
    ''' Error test
    ''' </summary>
    <Fact>
    Public Sub ErrorTest()
        Dim base64 = Convert.ToBase64String(New Byte() {1, 2, 3, 10, 11, 12, 13})
        StackLevel01(base64)
    End Sub
    Public Sub StackLevel01(base64 As String)
        StackLevel02(base64)
    End Sub
    Public Sub StackLevel02(base64 As String)
        StackLevel03(base64)
    End Sub
    Public Sub StackLevel03(base64 As String)
        Convert.FromBase64String(base64.Substring(1))
    End Sub

    ''' <summary>
    ''' Skipped Test
    ''' </summary>
    <Fact(Skip:="Skipped test demo")>
    Public Sub SkipTest()
    End Sub

    ''' <summary>
    ''' Fail Test
    ''' </summary>
    <Fact>
    Public Sub FailTest()
        Assert.True(IsPrime(411))
    End Sub

    Private Shared Function IsPrime(value As Integer) As Boolean
        Dim primeI As Integer
        Dim primeFlag = True
        For primeI = 2 To value / 2
            If value Mod primeI = 0 Then
                primeFlag = False
            End If
        Next
        Return primeFlag
    End Function

End Class

